### PR TITLE
Add new entity type: TreasuryAsset

### PR DESCRIPTION
**Pull request by cujowolf**
This new entity will track the individual performance and values of each asset held by the treasury. Token balance, RFV, Market Value are all calculated and stored within this entity. The sum total for the treasury is updated as a part of generating/updating the asset details.

One driving factor this this addition is the fact that we track multiple values for the same asset, not just dollar value. We are interested in the total tonnes of carbon held by the treasury for example.

### CODE REVIEW

File: src/utils/ProtocolMetrics.ts [hunk 1]
```diff
@@ -78,52 +96,108 @@ function getSklimaSupply(transaction: Transaction): BigDecimal {
...
+    let totalMarketValue = BigDecimal.fromString("0")
+
+    for (let i = 0; i < assets.length; i++) {
+        let assetDetail = loadOrCreateTreasuryAsset(transaction.timestamp, assets[i].substring(10))
+        totalCarbon = totalCarbon.plus(assetDetail.carbonBalance)
+        totalRFV = totalRFV.plus(assetDetail.riskFreeValue)
+        totalMarketValue = totalMarketValue.plus(assetDetail.marketValue)
+    }
+
+    //log.debug("Treasury Market Value {}", [mv.toString()])
```
Comment (0xAurelius) [inline src/utils/ProtocolMetrics.ts:189]: Do we want these debug logs to be left in here? Looks like one is still commented

File: src/utils/ProtocolMetrics.ts [hunk 2]
```diff
@@ -216,21 +290,25 @@ export function updateProtocolMetrics(transaction: Transaction): void {
     pm.sKlimaCirculatingSupply = getSklimaSupply(transaction)
 
     //KLIMA Price
-    pm.klimaPrice = getKLIMAUSDRate()
+    //pm.klimaPrice = getKLIMAUSDRate()
```
Comment (0xAurelius) [inline src/utils/ProtocolMetrics.ts:293]: Was this commented out deliberately?

**Review by 0xAurelius — APPROVED** [review]
Overall looks good, left a couple of minor questions

File: src/utils/ProtocolMetrics.ts (hunk 2 above)
Comment (cujowolf) [reply src/utils/ProtocolMetrics.ts:293]: It was. Currently there is a divide by 0 error when attempting to index from the starting block defined in subgraph.yaml. Was going add the price function updates as their own commit and un comment at that time.

File: src/utils/ProtocolMetrics.ts (hunk 1 above)
Comment (cujowolf) [reply src/utils/ProtocolMetrics.ts]: This is actually duplicated on line 192. Removed the commented duplicate log message.

**Pull request merged**